### PR TITLE
Add FXIOS-9646 - Password Generator Telemetry

### PR DESCRIPTION
**Pull request by DanielDervishi**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-9646)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/21248)

## :bulb: Description

Add 2 metrics to telemetry:
- Password Generator Prompt Shown: Counter
- Password field filled with generated password: Counter

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [x] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

**Review by nbhasin2 — APPROVED** [review]

File: firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorTelemetry.swift [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+public struct PasswordGeneratorTelemetry {
+    func passwordGeneratorDialogShown() {
+        GleanMetrics.PasswordGenerator.shown.add()
+    }
+
+    func usePasswordButtonPressed() {
+        GleanMetrics.PasswordGenerator.filled.add()
+    }
+}
```
Comment (nbhasin2) [inline firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorTelemetry.swift:16]: Where do you plan to use this?

Comment (DanielDervishi) [reply firefox-ios/Client/Frontend/PasswordGenerator/PasswordGeneratorTelemetry.swift:16]: Not too sure, just waiting on product to get back to me to figure out how the data would be used. Once I hear back from them I'll fill out the data review form and tag Roux :)

**Comment by DanielDervishi** [comment]
# Request for data collection review form
**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

1) What questions will you answer with this data?
$\color{red}{\textsf{- Number of times users open the password generator prompt}}$
$\color{red}{\textsf{- Number of times users fill in the password field with the generated password}}$

2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements? Some example responses:
$\color{red}{\textsf{- Track user adoption}}$

3) What alternative methods did you consider to answer these questions? Why were they not sufficient?
$\color{red}{\textsf{- This is a new feature and so we wouldn't have any other metrics to infer from}}$

4) Can current instrumentation answer these questions?
$\color{red}{\textsf{- No, this is a new feature}}$

5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Data_Collection) found on the Mozilla wiki.   

**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>Password Generator: Prompt Shown</td>
    <td>Category 2</td>
    <td>#21248</td>
  </tr>
  <tr>
      <td>Password Generator: Generated Password Used</td>
      <td>Category 2</td>
      <td>#21248</td>
    </tr>
</table>

6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way.
- This collection is documented in the Glean Dictionary at https://dictionary.telemetry.mozilla.org/
 
7) How long will this data be collected?  Choose one of the following:
$\color{red}{\textsf{- I want this data to be collected for 6 months initially (potentially renewable).}}$

8) What populations will you measure?
$\color{red}{\textsf{ - Only release - All countries}}$

9) If this data collection is default on, what is the opt-out mechanism for users?
$\color{red}{\textsf{- These collections are Glean. The opt-out can be found in Settings}}$

10) Please provide a general description of how you will analyze this data.
$\color{red}{\textsf{- Feature Usage/Adoption. Improve user flows}}$

11) Where do you intend to share the results of your analysis?
$\color{red}{\textsf{- Internal Product Team}}$

12) Is there a third-party tool (i.e. not Glean or Telemetry) that you are proposing to use for this data collection? If so:
$\color{red}{\textsf{- No}}$

@adudenamedruby

**Comment by adudenamedruby** [comment]
r+
# Data Review Form

_1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?_ 

Yes, telemetry is documented at https://dictionary.telemetry.mozilla.org/apps/firefox_ios

_2) Is there a control mechanism that allows the user to turn the data collection on and off?_ 

Yes, Firefox iOS provides data controls in the app settings

_3) If the request is for permanent data collection, is there someone who will monitor the data over time?_

This will be reviewed by a data-steward on the Firefox iOS team (@adudenamedruby) every 6 months

_4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?_

Type 2

_5) Is the data collection request for default-on or default-off?_

Default on

_6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?_

No

_7) Is the data collection covered by the existing Firefox privacy notice?_ 

Yes

_8) Does the data collection use a third-party collection tool?_ 

No

**Pull request merged**